pkg/services/ngalert/state/historian/backend.go

```diff
@@ -0,0 +1,39 @@
+package historian
+
+import (
+	"fmt"
+	"strings"
+)
+
+// BackendType identifies different kinds of state history backends.
+type BackendType string
+
+// String implements Stringer for BackendType.
+func (bt BackendType) String() string {
+	return string(bt)
+}
+
+const (
+	BackendTypeAnnotations BackendType = "annotations"
+	BackendTypeFanout      BackendType = "fanout"
+	BackendTypeLoki        BackendType = "loki"
+	BackendTypeNoop        BackendType = "noop"
+	BackendTypeSQL         BackendType = "sql"
+)
```

Comment:
I thought you would put this enum to the settings package and will validate the mode when it parses settings which will make Grafana crash faster, during settings parsing. But I am ok with this too.

Reply:
Other consistency checks (are required keys for each backend provided? are urls valid? etc) also already happen in ngalert package. I added it there to keep all validation in one place.

I'm not opposed to moving the validation as a whole to config package, though! I'd just rather not split validation for now